photon-lib/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -100,22 +100,22 @@ case class Coefficients(means: Vector[Double], variancesOption: Option[Vector[Do
    * @param that The other Coefficients to compare to
    * @return True if the Coefficients are equal, false otherwise
    */
-  override def equals(that: Any): Boolean =
-    that match {
-      case other: Coefficients =>
-        val (m1, v1, m2, v2) = (this.means, this.variancesOption, other.means, other.variancesOption)
-        val sameType = m1.getClass == m2.getClass && v1.map(_.getClass) == v2.map(_.getClass)
-        lazy val sameMeans = VectorUtils.areAlmostEqual(m1, m2)
-        lazy val sameVariance = (v1, v2) match {
-          case (None, None) => true
-          case (Some(val1), Some(val2)) => VectorUtils.areAlmostEqual(val1, val2)
-          case (_, _) => false
-        }
-
-        sameType && sameMeans && sameVariance
-
-      case _ => false
-    }
+  override def equals(that: Any): Boolean = that match {
+    case other: Coefficients =>
+      val (m1, v1, m2, v2) = (this.means, this.variancesOption, other.means, other.variancesOption)
+      val sameType = (m1.getClass == m2.getClass) && (v1.map(_.getClass) == v2.map(_.getClass))
```

Comment:
I recall we had a conversation about a bug being caused by this line, where the classes were expected to match but didn't need to. Is that not that case?

Reply:
No. The bug is in https://github.com/linkedin/photon-ml/pull/445/files/d3c24eb0a934377784bc2dea1bf82e9cb308007b#diff-b0fe65697761c5d596b4067d05825968L36, where the type of the means should match the type of variance.